eras/conway/impl/src/Cardano/Ledger/Conway/UTxO.hs

```diff
@@ -57,22 +72,59 @@ getConwayScriptsNeeded utxo txb = getAlonzoScriptsNeeded utxo txb <> voterScript
         txb ^. votingProceduresTxBodyL
 
 conwayProducedValue ::
-  ConwayEraTxBody era =>
+  ( ConwayEraTxBody era
+  , ConwayEraTxCert era
+  , ConwayEraPParams era
+  ) =>
   PParams era ->
   (KeyHash 'StakePool (EraCrypto era) -> Bool) ->
+  (Credential 'DRepRole (EraCrypto era) -> Maybe Coin) ->
   TxBody era ->
   Value era
-conwayProducedValue pp f txb =
+conwayProducedValue pp isStakePool lookupDRepDeposit txb =
   inject propDeposits
     <> inject (txb ^. treasuryDonationTxBodyL)
-    <> shelleyProducedValue pp f txb
+    <> shelleyProducedValue pp isStakePool lookupDRepDeposit txb
+    <> inject drepDeposits
   where
     propDeposits = foldMap pProcDeposit $ txb ^. proposalProceduresTxBodyL
+    drepDeposits =
+      let go accum@(!drepRegsInTx, !totalDeposit) = \case
+            RegDRepTxCert c _ _
+              -- Re-registration is a no-op and not considered
+              | Just _ <- lookupDRepDeposit c -> accum
+              | otherwise -> (Set.insert c drepRegsInTx, totalDeposit <+> pp ^. ppDRepDepositL)
+            UnRegDRepTxCert c _
+              -- DRep previously registered in the same tx.
+              | Set.member c drepRegsInTx -> (Set.delete c drepRegsInTx, totalDeposit <-> pp ^. ppDRepDepositL)
+            _ -> accum
+       in snd $ foldl' go (Set.empty, Coin 0) $ txb ^. certsTxBodyL
```

Comment:
This is incorrect. You totally mixed it up. Produced is what leaves the transaction (outputs, fees and deposits). While consumed is what adds to the transaction (inputs, withdrawals and refunds)

In this PR you are doing the opposite.

Implement it by example in the same way the Stake Delegation deposits are implemented here: 

* for produced: https://github.com/input-output-hk/cardano-ledger/blob/d14ff95a51e7c229cc64a07579976d9a2bdd9078/eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/RefundsAndDeposits.hs#L60
* for consumed: https://github.com/input-output-hk/cardano-ledger/blob/d14ff95a51e7c229cc64a07579976d9a2bdd9078/eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/RefundsAndDeposits.hs#L118C1-L118C16

Reply:
Thanks for the pointers! :bowing_man: